src/generator/writeProgram.ts

```diff
@@ -269,10 +269,12 @@ function writeAll(ctx: CompilerContext, wctx: WriterContext, name: string, abiLi
     for (let t of sortedTypes) {
         if (t.kind === 'contract' || t.kind === 'struct') {
             let allocation = getAllocation(ctx, t.name);
+            let allocationBounced = getAllocation(ctx, toBounced(t.name));
```

Comment:
Finalize how allocation should work - 
Currently the type store relies on the partialFields property,
whereas the allocator pushes a separate bounced type to the array, which doesn't feel right.